### ISSUE TITLE
[AXE-CORE]: ARIA dialog and alertdialog nodes should have an accessible name

### ISSUE THREAD

**Issue by alexandec** [department-of-veterans-affairs/va.gov-team#43444]
- [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

## User Story or Problem Statement
As a user with visual impairment, I expect to the signin modal to be clearly labeled so I can determine the purpose of the modal.

## Details
The signin modal on va.gov refers to a label element as follows:

```
aria-labelledby="signin-signup-modal-title"
```

The `signin-signup-modal-title` element referred to here does not exist, so the modal is not properly labelled.

Axe DevTools detects this error and says, "Ensure every ARIA dialog and alertdialog node has an accessible name".

## Acceptance Criteria

- [ ] The signin modal on va.gov has a proper ARIA label

## Environment

* Operating System: MacOS 12.4 (21F79), Monterey
* Browser: Chrome 102.0.5005.115 (Official Build) (x86_64)
* Server: Production

## Steps to Recreate

1. Navigate to va.gov
2. Click the "Sign in" button
3. Inspect the page with Dev Tools and search for `signin-signup-modal-title`
4. Note that the modal refers to this element, but the element is not present
5. Run Axe DevTools and do a full page scan with the modal open. Axe will report the issue

## Proposed Solution (if known)

Add a `signin-signup-modal-title` element which properly labels the modal. It's possible this element previously existed and was removed at some point.

## WCAG or Vendor Guidance (optional)

- https://dequeuniversity.com/rules/axe/4.4/aria-dialog-name

**Comment by it-harrison** [comment]
[PR here](https://github.com/department-of-veterans-affairs/vets-website/pull/23006)
